src/include/units/quantity.h

```diff
@@ -322,18 +322,18 @@ template<typename D, typename U1, typename Rep1, typename U2, typename Rep2>
   return ret(ret(lhs).count() - ret(rhs).count());
 }
 
-template<typename D, typename U, typename Rep, Scalar Value>
+template<typename D, typename U, typename Rep, DimensionlessQuantity Value>
```

Comment:
Did you mean Arithmetic here? I do not think that in `2m * 2` a `2` is a dimensionless quantity... It is just a regular value/multiplier/ratio or however you want to call it (if not a scalar).

Reply:
> Did you mean Arithmetic here? 

No I didnt.  T :  model of DimensionlessQuantity  requires only
T x ;
Dimensionless y = x * x;
(note: eliding division for now)
Usually the following is true,  e.g for arithmetic
T x ;
x = x * x;
( but not necessarily which allows dimensionless ratios such as radian and steradian) 

> I do not think that in `2m * 2` a `2` is a dimensionless quantity...

If you are asking :  is the 2 in the above a DimensionlessQuantity?, **_Yes it is a dimensionless quantity_** , plugging the above requirement to an integer 
int x = 2;
int y = x * x;    // works fine

Scalar is a different concept to DimensionlessQuantity. A Scalar has magnitude but not direction
https://www.britannica.com/science/scalar 

c++ arithmetic type is a model of Scalar and a model of DimensionlessQuantity
 **units::quantity with a Scalar Rep type is a model of Scalar but not a model of DimensionlessQuantity**. 

You could examine the rep type of the units::quantity to find if the quantity is  a Scalar. If the Rep type is a scalar, then the units::quantity is a scalar, but If the Rep type is a vector or a matrix then clearly  the quantity isn't a Scalar